### DISCUSSION TITLE
contributing/development/compiling/optimizing_for_size

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#154] [User-contributed notes]
# contributing/development/compiling/optimizing_for_size

Rationale: Sometimes, it is desired to optimize a build for size rather than speed. This means not compiling unused functions from the engine, as well as using specific compiler flags to aid on dec...

https://docs.godotengine.org/en/stable/contributing/development/compiling/optimizing_for_size.html

**Comment by ghost** [comment]
For a web file there is one more possible optimization for size: `wasm-opt` tool from [binaryen](https://github.com/WebAssembly/binaryen). Can be used as `wasm-opt <original.wasm> -o <optimized.wasm> -all --post-emscripten -Oz`

* `-Oz` means size optimization
* `--post-emscripten` as name say optimizations for files produced by emscripten
* `-all` enables support for all features declared in the wasm file. Without it you will need to enable particular features manually which can differ from version to version of godot...

Does it have any performance penalties? Usually optimizing for size mean performance penalties so I think yes.
Can it output something which doesn't work? Idk, maybe...

**Reply by Arsen-Lovecraft** [reply]
I saved 2.7 mb on custom build. From 25.7 to 23.0.

**Reply by Swarkin** [reply]
22.4 MiB -> 20.0 MiB, thanks!

**Reply by Calinou** [reply]
Thanks for the information! jordi opened a pull request based on this comment to automatically run wasm-opt at build-time: https://github.com/godotengine/godot/pull/97457

**Comment by Swarkin** [comment]
Here's a good read about the topic: https://popcar.bearblog.dev/how-to-minify-godots-build-size/
